### ISSUE TITLE
Implement support for RFC 8, transitive NPM dependencies

### ISSUE THREAD

**Issue by alexcrichton** [rustwasm/wasm-pack#606]
Specified upstream [RFC 8](https://github.com/rustwasm/rfcs/blob/master/text/008-npm-dependencies.md) is a way for NPM dependencies to work transitively throughout the crate graph. The first phase of supporting this is [now implemented in wasm-bindgen](https://github.com/rustwasm/wasm-bindgen/pull/1305) and the next phase is in `wasm-pack` itself. Thankfully it shouldn't be too hard!

Currently today `wasm-pack` will write out a `package.json`, but this will overwrite a `package.json` created by `wasm-bindgen`, if one exists. Instead what `wasm-pack` should do is:

* When creating `package.json`, first check to see if one already exists at the target location
* Deserialize the `dependencies` map, if it exists, otherwise start with an empty object
* Add in all the current configuration (name, scope, urls, etc)
* Write out the `package.json` again, overwriting what was previously there.

Basically the `package.json` that `wasm-pack` writes should just be merged with the `package.json` emitted by `wasm-bindgen` if it's already there.

**Comment by Tarnadas** [comment]
Hey @alexcrichton,

Let me know what you think about my PR #732

**Comment by trivigy** [comment]
The restriction in the RFC which restricts package.json to only one key `dependencies` builds the entire build process because of [THIS](https://github.com/rustwasm/rfcs/pull/8#issuecomment-564725214). Would greatly appreciate some advice on how to work around this restriction.
